src/lua/uri.lua

```diff
@@ -173,6 +241,57 @@ local function format(uri, write_password)
     return str
 end
 
+-- Encodes a string into its escaped hexadecimal representation.
+local function escape(buf, opts)
+    if type(buf) ~= "string" then
+        error("Usage: uri.escape(string, [opts])")
+    end
+    if opts and type(opts) ~= "table" then
+        error("opts must be a table")
+    end
+    assert(RFC3986.plus == false)
```

Comment:
This looks like "static_assert", but it is checked in runtime to prevent tweaks in RFC3986 table. Sounds reasonable, but I suggest to drop this reason in the comment nearby, so "future generations" will not remove this assertion later.